packages/gibsonify_api/lib/src/models/measurement.dart

```diff
@@ -90,36 +110,49 @@ class Measurement extends Equatable {
     r'^\d+$', // numeric only
   );
 
+  List<String> getMeasurementUnitsForMethod() {
+    if (method == null) {
+      return ['Choose a measurement method first'];
+    } else {
+      return Measurement.measurementUnitsOfMethod[method] ??
+          ['No units for given method'];
+    }
+  }
+
```

Comment:
Not sure about this, imo the units dropdown should just be disabled if the method has not been selected.

Reply:
@adam-sroka @sshazril i agree with this